magenta-lib/src/main/scala/magenta/tasks/FastlyComputeTasks.scala

```diff
@@ -49,12 +51,40 @@ case class FastlyComputeTasks(s3Package: S3Path)(implicit
         stopFlag
       )
 
-      activateVersion(nextVersionNumber, client, resources.reporter, stopFlag)
+      activateVersion(
+        nextVersionNumber,
+        client,
+        resources.reporter,
+        stopFlag
+      )
+
+      nextVersionNumber
+    }
+  }
 
-      resources.reporter
-        .info(
-          s"Fastly Compute@Edge service ${client.serviceId} - version $nextVersionNumber is now active"
+  override def execute(
+      resources: DeploymentResources,
+      stopFlag: => Boolean
+  ): Unit = {
+    FastlyApiClientProvider
+      .get(keyRing)
+      .map(client =>
+        (client, executeWithCredentials(client, resources, stopFlag))
+      ) match {
+      case None =>
+        resources.reporter.fail(
+          "Failed to fetch Fastly API credentials from keyring"
+        )
+      case Some((client, Success(nextVersionNumber))) =>
+        resources.reporter
+          .info(
+            s"Fastly Compute@Edge service ${client.serviceId} - version $nextVersionNumber is now active"
+          )
+      case Some((_, Failure(err))) =>
+        resources.reporter.fail(
+          s"$err. Your API key may be invalid or it may have expired"
```

Comment:
Minor: 

IIUC, this case is going to catch a number of potential errors (network problems, Fastly API downtime/errors) as well as the problems that we've already mentioned in the error message. 

If we find reliability problems with this deployment type in the future, it might be worth spending a bit of extra time to extract the specific error (perhaps by adding some verbose logging, or a different return type to [`activateVersion`](https://github.com/guardian/riff-raff/blob/1522cee3f148bb279099f686fa62afcf18383f4f/magenta-lib/src/main/scala/magenta/tasks/FastlyComputeTasks.scala#L141-L153)), but I don't think that needs to block this PR.

Reply:
Thinking about this some more, I think there is another potential false positive here. IIUC calling [`activateVersion`](https://github.com/guardian/riff-raff/blob/1522cee3f148bb279099f686fa62afcf18383f4f/magenta-lib/src/main/scala/magenta/tasks/FastlyComputeTasks.scala#L141-L153) can result in a successful `Future` (i.e. we got _a_ `Response` from the Fastly API), where the response/status code indicates that the request was unsuccessful from Fastly's perspective (i.e. we got a 500 back and did not actually activate the new version). In this case we will still match on:  ```case Some((client, Success(nextVersionNumber)))```.

Happy to pair on fixing that via a separate PR if helpful.